community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/schema/SchemaKernelException.java

```diff
@@ -50,20 +53,29 @@ public SchemaKernelException( Status statusCode, String message )
         super( statusCode, message );
     }
 
-    protected static String messageWithLabelAndPropertyName( TokenNameLookup tokenNameLookup, String formatString,
+    static String messageWithLabelAndPropertyName( TokenNameLookup tokenNameLookup, String formatString,
             LabelSchemaDescriptor descriptor )
     {
+        int[] propertyIds = descriptor.getPropertyIds();
+
         if ( tokenNameLookup != null )
         {
+            String propertyString = propertyIds.length == 1 ?
```

Comment:
Yes probably, for the error message to have the correct pluralization we still need to check the length of the array though.